### PR TITLE
Ties placement and hiding

### PR DESCRIPTION
**Pull request by alexpavlov96**

### CODE REVIEW

File: src/engraving/dom/note.cpp [hunk 1]
```diff
@@ -1378,6 +1378,72 @@ bool Note::isNoteName() const
     return false;
 }
 
+bool Note::shouldHideFret() const
+{
+    if (!tieBack() || shouldForceShowFret()) {
+        return false;
+    }
+
+//    if (isContinuationOfBend()) {
```
Comment (mikekirin) [inline src/engraving/dom/note.cpp:1387]: Please delete this or leave a comment

File: src/engraving/dom/shape.cpp [hunk 2]
```diff
@@ -297,6 +297,30 @@ double Shape::bottom() const
...
+    }
+
+    return edge;
+}
+
+double Shape::leftMostEdgeAtHeight(double yAbove, double yBelow) const
+{
+    double edge = 100000.0;
+    for (const ShapeElement& sh : *this) {
+        if (sh.bottom() > yAbove && sh.top() < yBelow) {
```
Comment (mikekirin) [inline src/engraving/dom/shape.cpp:316]: Is there any possibility we'll never pass this condition? If so how 100000.0 value will affect the layout/drawing?

File: src/engraving/dom/shape.cpp [hunk 3]
```diff
@@ -297,6 +297,30 @@ double Shape::bottom() const
     return dist;
 }
 
+double Shape::rightMostEdgeAtHeight(double yAbove, double yBelow) const
+{
+    double edge = -100000.0;
+    for (const ShapeElement& sh : *this) {
+        if (sh.bottom() > yAbove && sh.top() < yBelow) {
```
Comment (mikekirin) [inline src/engraving/dom/shape.cpp:304]: Is there any possibility we'll never pass this condition? If so how -100000.0 value will affect the layout/drawing?

File: src/engraving/rendering/dev/slurtielayout.cpp [hunk 4]
```diff
@@ -987,10 +991,45 @@ void SlurTieLayout::fixArticulations(Slur* item, PointF& pt, Chord* c, double up
...
+                item->setUp(c1->up());
+            } else {
+                item->setUp(!c1->up());
+            }
+        }
+    } else {
+        item->setUp(item->slurDirection() == DirectionV::UP ? true : false);
+    }
+
+    item->fixupSegments(1);
```
Comment (mikekirin) [inline src/engraving/rendering/dev/slurtielayout.cpp:1016]: What does 1 means?

File: src/engraving/rendering/dev/slurtielayout.cpp [hunk 5]
```diff
@@ -1000,11 +1039,28 @@ static bool tieSegmentShouldBeSkipped(Tie* item)
         return false;
     }
 
-    return !st->showBackTied() || (startNote && startNote->harmonic());
+//    if (startNote->isContinuationOfBend()) {
```
Comment (mikekirin) [inline src/engraving/rendering/dev/slurtielayout.cpp:1042]: Please remove or add a comment

File: src/engraving/dom/shape.cpp (hunk 3 above)
Comment (RomanPudashkin) [reply src/engraving/dom/shape.cpp:304]: @mikekirin https://github.com/musescore/MuseScore/pull/20656

File: src/engraving/dom/shape.cpp (hunk 2 above)
Comment (mike-spa) [reply src/engraving/dom/shape.cpp:316]: see #20656, @alexpavlov96 you may want to pick that commit too

File: src/engraving/rendering/dev/slurtielayout.cpp (hunk 4 above)
Comment (mike-spa) [reply src/engraving/rendering/dev/slurtielayout.cpp:1016]: It means that the Tie only needs 1 TieSegment, which is correct in this case

**Review by mike-spa — APPROVED** [review]

**Review by mikekirin — APPROVED** [review]

**Pull request merged**